changelog/add-migrate-horizontal-list-to-ts

```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: dev
+
+Migrate `HorizontalList` component to TS
```

Comment:
I don't think we should be including changelog entries for this since it's not relevant for merchants or anyone reading the plugin changelog.

Unless `dev` type changelog messages don't appear in the actual changelog? 🤔 

I'm unsure on the policy here so I've asked in p1684186578137329-slack-CGGCLBN58.